src/audio/pipeline/pipeline-stream.c

```diff
@@ -55,7 +55,8 @@ pipeline_should_report_enodata_on_trigger(struct comp_dev *rsrc,
 	/* if component on which we depend to provide data is inactive, then the
 	 * pipeline has no means of providing data
 	 */
-	if (rsrc->state != COMP_STATE_ACTIVE)
+	if (rsrc->pipeline->priority > ppl_data->start->pipeline->priority &&
```

Comment:
Can you update the comment too and explain the decision logic. Thanks

Reply:
sure, thanks